src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -705,33 +713,76 @@ const WorkflowView = _.flow(
+                      })
+                    } else {
+                      this.setState(_.set(['modifiedConfig', 'rootEntityType'], value))
+                      this.setState(_.unset(['modifiedConfig', 'dataReferenceName']))
+                      this.setState({
+                        selectedEntityType: value, selectedTableName: undefined,
+                        entitySelectionModel: this.resetSelectionModel(value, {}, entityMetadata, source), selectedEntitySource: source
+                      })
+                    }
                   },
-                  options: [...entityTypes, ...possibleSetTypes].sort()
+                  options: [
+                    {
+                      label: 'TABLES',
+                      options: _.map(entityType => ({ value: entityType, source: 'table' }), [...entityTypes, ...possibleSetTypes].sort())
+                    },
+                    ...(!_.isEmpty(availableSnapshots) ? [{
+                      label: 'SNAPSHOTS',
+                      options: _.map(({ name }) => ({ value: name, source: 'snapshot' }), availableSnapshots)
+                    }] : [])
+                  ]
                 })
               ]),
-              div({ style: { marginLeft: '2rem', paddingLeft: '2rem', borderLeft: `2px solid ${colors.dark(0.2)}`, flex: 1 } }, [
-                div({ style: { height: '2rem', fontWeight: 'bold' } }, ['Step 2']),
-                div({ style: { display: 'flex', alignItems: 'center' } }, [
-                  h(ButtonPrimary, {
-                    disabled: currentSnapRedacted || this.isSingle() || !rootEntityType ||
-                      !_.includes(selectedEntityType, [...entityTypes, ...possibleSetTypes]) || !!Utils.editWorkspaceError(ws),
-                    tooltip: Utils.editWorkspaceError(ws),
-                    onClick: () => this.setState({ selectingData: true })
-                  }, ['Select Data']),
-                  label({ style: { marginLeft: '1rem' } }, [`${this.describeSelectionModel()}`])
+              (entitySelectionModel.type === processSnapshotTable) ? h(IdContainer, [id => div({ style: { margin: '2rem 0 0 2rem' } }, [
+                h(Select, {
+                  id,
+                  isDisabled: !!Utils.editWorkspaceError(ws),
+                  isClearable: false,
+                  value: selectedTableName,
+                  onChange: ({ value }) => {
+                    this.setState({ selectedTableName: value })
+                    this.setState(_.set(['modifiedConfig', 'rootEntityType'], value))
+                    this.setState(_.unset(['modifiedConfig', 'entityName']))
+                  },
+                  styles: { container: old => ({ ...old, display: 'inline-block', width: 200, marginLeft: '0.5rem' }) },
+                  options: _.sortBy(_.identity)(selectedSnapshotTableNames)
```

Comment:
nit: if you're passing multiple params at once to a curried lodash function, you can just make a single call, e.g. `_.sortBy(_.identity, selectedSnapshotTableNames)`